xidmap/xidmap.go

```diff
@@ -94,6 +92,14 @@ func New(zero *grpc.ClientConn, db *badger.DB) *XidMap {
 			tree: z.NewTree(100 << 20),
 		}
 	}
+
+	xm.kvChan = make(chan []kv, 64)
+
+	for i := 0; i < 16; i++ {
+		xm.wg.Add(1)
+		go xm.dbWriter()
```

Comment:
The db could be nil. You should start these go routines inside the following if block